python/sglang/lang/interpreter.py

```diff
@@ -411,24 +411,47 @@ def _execute_role_end(self, expr: SglRoleEnd):
         self._execute_fill(suffix)
 
         if self.cur_images:
-            # OpenAI vision API format
-            last_msg = {
-                "role": expr.role,
-                "content": [{"type": "text", "text": new_text}],
-            }
-            for image_path, image_base64_data in self.cur_images:
-                last_msg["content"].append(
-                    {
-                        "type": "image_url",
-                        "image_url": {
-                            "url": f"data:image/jpeg;base64,{image_base64_data}"
-                        },
-                    }
-                )
+            if hasattr(self.backend, "name") and self.backend.name == "gemini":
+                # Gemini vision API format
+                last_msg = {
+                    "role": expr.role,
+                    "parts": [{"text": new_text}],
+                }
+                for image_path, image_base64_data in self.cur_images:
+                    last_msg["parts"].append(
+                        {
+                            "inline_data": {
+                                "data": image_base64_data,
+                                "mime_type": "image/jpeg",
+                            },
+                        }
+                    )
```

Comment:
It is better to use same format with openai and have an internal format translation for Gemini. Because we want the user to have a unified interface to get messages, see example below:
https://github.com/sgl-project/sglang/blob/ffe4aaee1da5ccfdda1c6228ae911d6f139b68eb/examples/quick_start/openai_example_chat.py#L19-L20